src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,88 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_SIZE = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K,V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_SIZE];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        grow();
+        int keyHash = getHash(key);
+        if (table[keyHash % table.length] == null) {
+            table[keyHash % table.length] = new Node<>(key, value, keyHash, null);
+            size++;
+        } else {
+            Node<K, V> curNode = table[keyHash % table.length];
+            while (curNode != null) {
+                if (key == null ? curNode.key == null : key.equals(curNode.key)) {
```

Comment:
In this task you may use Object.equals method for simplify and shortening your code. Mind, there is no comments about it in list of common mistakes.